providers/os/resources/discovery/container_registry/resolver.go

```diff
@@ -50,12 +47,11 @@ func (r *Resolver) Resolve(ctx context.Context, root *inventory.Asset, conf *inv
 	if err == nil {
 		log.Debug().Str("image", conf.Host).Msg("detected container image in container registry")
 
-		remoteOpts := AuthOption(conf.Credentials, credsResolver)
-		// we need to disable default keychain auth if an auth method was found
-		if len(remoteOpts) > 0 {
```

Comment:
this was redundant, if we ensure that the auth opt always returns then we dont need the `DisableKeychainAuth` since we can have the `AuthOption` decide whats the best auth method by looking at the creds